internal/agent/upgrade.go

```diff
@@ -117,24 +121,51 @@ func Upgrade(
 
 	utils.SetEnv(c.Env)
 
-	args := []string{"upgrade", "--system.uri", fmt.Sprintf("docker:%s", img)}
-	args = append(args,
-		"--auth-username", authUser,
-		"--auth-password", authPass,
-		"--auth-server-address", authServer,
-		"--auth-type", authType,
-		"--auth-registry-token", registryToken,
-		"--auth-identity-token", identityToken,
-	)
-
+	// Load the upgrade Config from the system
+	resetConfig, err := elementalConfig.ReadConfigRun("/etc/elemental")
```

Comment:
This variable should probably be called `upgradeConfig` (though it seems that we load the same config on reset so maybe a more generic name in both cases would be better)

Reply:
indeed. It actually should be config but there is already like 32 different config vars lol. I'll change it!

Reply:
Yes, we will hopefully consolidate configs [here](https://github.com/kairos-io/kairos/issues/1225) (That story is a can of worms...)